controllers/drplacementcontrol_controller.go

```diff
@@ -1602,65 +1742,224 @@ func (s ObjectStorePVDownloader) DownloadPVs(ctx context.Context, r client.Reade
 }
 
 func (d *DRPCInstance) advanceToNextDRState() {
-	var nextState rmn.DRState
+	lastDRState := d.getLastDRState()
+	nextState := lastDRState
 
-	switch d.instance.Status.LastKnownDRState {
-	case rmn.DRState(""):
-		nextState = rmn.Initial
+	switch lastDRState {
+	case rmn.Deploying:
+		nextState = rmn.Deployed
 	case rmn.FailingOver:
 		nextState = rmn.FailedOver
 	case rmn.FailingBack:
 		nextState = rmn.FailedBack
 	case rmn.Relocating:
 		nextState = rmn.Relocated
-	case rmn.Initial:
+	case rmn.Deployed:
 	case rmn.FailedOver:
 	case rmn.FailedBack:
 	case rmn.Relocated:
-	default:
-		nextState = rmn.DRState("")
 	}
 
 	d.setDRState(nextState)
 }
 
-func (d *DRPCInstance) resetDRState() {
-	d.log.Info("Resetting last known DR state", "lndrs", d.instance.Status.LastKnownDRState)
+func (d *DRPCInstance) setDRState(nextState rmn.DRState) {
+	if d.instance.Status.Phase != nextState {
+		d.log.Info(fmt.Sprintf("Phase: Current '%s'. Next '%s'",
+			d.instance.Status.Phase, nextState))
+
+		d.instance.Status.Phase = nextState
+		d.updateConditions()
+		d.needStatusUpdate = true
+	}
+}
+
+func (d *DRPCInstance) updateConditions() {
+	d.log.Info(fmt.Sprintf("Current Conditions '%v'", d.instance.Status.Conditions))
+
+	for _, condType := range []string{rmn.ConditionAvailable, rmn.ConditionReconciling} {
+		condition := d.newCondition(condType)
 
-	d.setDRState(rmn.DRState(""))
+		idx, _ := GetDRPCCondition(&d.instance.Status, condType)
+		if idx == -1 {
+			d.instance.Status.Conditions = append(d.instance.Status.Conditions, *condition)
+		} else {
+			d.instance.Status.Conditions[idx] = *condition
```

Comment:
This may overwrite a message that is set in the condition. I am thinking on how to get relevant messages from other parts of the reconcile and preserve it when updating here.